src/main/java/com/google/devtools/build/lib/remote/options/RemoteOptions.java

```diff
@@ -483,6 +483,17 @@ public RemoteOutputsStrategyConverter() {
               + "that loads objects from the CAS on demand.")
   public String remoteDownloadSymlinkTemplate;
 
+  @Option(
+          name = "experimental_remote_deduplicate_uploads",
+          defaultValue = "false",
+          category = "remote",
+          documentationCategory = OptionDocumentationCategory.REMOTE,
+          effectTags = {OptionEffectTag.EXECUTION},
+          help =
+              "If set to true, Bazel deduplicates calls to the remote service to find missing "
+                  + "files and also deduplicates uploads.")
+  public boolean experimentalRemoteDeduplicateUploads;
```

Comment:
We have too much flags now. Are there concerns to introduce this flag and disable the feature by default? I think people always want this enabled.